logic.js

```diff
@@ -0,0 +1,136 @@
+const Tweeter = function () {
+    let posts = [{
+            text: "First post!",
+            id: "p1",
+            comments: [{
+                    id: "c1",
+                    text: "First comment on first post!"
+                },
+                {
+                    id: "c2",
+                    text: "Second comment on first post!!"
+                },
+                {
+                    id: "c3",
+                    text: "Third comment on first post!!!"
+                }
+            ]
+        },
+        {
+            text: "Aw man, I wanted to be first",
+            id: "p2",
+            comments: [{
+                    id: "c4",
+                    text: "Don't wory second poster, you'll be first one day."
+                },
+                {
+                    id: "c5",
+                    text: "Yeah, believe in yourself!"
+                },
+                {
+                    id: "c6",
+                    text: "Haha second place what a joke."
+                }
+            ]
+        }
+    ]
+
+    let postIdCounter = 0,
+        commentIdCounter = 0
+
+    const getPosts = function () {
+        return posts
+    }
+
+    const addPost = function (text) {
+        postIdCounter = posts.length + 1
+        let nextPostID = "p" + postIdCounter
+
+        const newPost = {
+            text: text,
+            id: nextPostID,
+            comments: []
+        }
+        posts.push(newPost)
+    }
+
+    const removePost = function (postID) {
+        posts.splice(postID, 1)
```

Comment:
The `postID` is not an index, you need to loop the array and check if it's equal to each id of a post.